graphs/possible_bipartition.py

```diff
@@ -1,12 +1,46 @@
-# Can be used for BFS
-from collections import deque 
+from collections import defaultdict  
 
 def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(V+E)
+        Space Complexity: O(V+E)
     """
-    pass
+    # create an adjacency list to represent the graph
+    graph = defaultdict(list)
+    for dislike in dislikes:
+        # if there are no enemies, no need to add to graph
+        if len(dislike) > 1:
+            dog, enemy = dislike[0], dislike[1]
+            graph[dog].append(enemy)
+        # if there are 3 or more dogs who fight each other, 
+        # you can't divide the non-fighters into just 2 groups,
```

Comment:
You are correct that if 3 or more dogs dislike each other, you can't divide them into just two happy groups. However, note that dog A can dislike a dog B without dog B disliking dog A (this is true in my example above). In other words, dogs don't have to _mutually_ dislike each other.

For this reason, we don't always want to return `False` when `len(dislike) > 2`